### PR TITLE
Remove KRA CLI XML options

### PR DESCRIPTION
**Pull request by ckelleyRH**
* Only allow JSON
* Provide JSON templates

### CODE REVIEW

**Comment by ckelleyRH** [comment]
Backport from master, there was a spurious conflict in the Tools documentation which I have ignored.

File: docs/changes/v11.1.0/Tools-Changes.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+== pkispawn and pkidestroy Changes ==
+
+Previously `pkispawn` and `pkidestroy` generated the logs on the console and also stored the logs into a file.
+The tools have been modified to only generate the logs on the console by default.
+To store the logs into a file as well, specify the file path using the `--log-file` option:
+
+----
+$ pkispawn ... --log-file <path>
+----
+
```
Comment (edewata) [inline docs/changes/v11.1.0/Tools-Changes.adoc:12]: I think this file should be moved into a `v11.0.3` folder since that would be the next version to be released from the `v11.0` branch. Also the "pkispawn and pkidestroy Changes" section should not be included since those changes were done in master (i.e. PKI 11.1) branch.

File: tests/dogtag/acceptance/cli-tests/pki-key-cli/pki-key-cli-template-show-kra.sh [hunk 2]
```diff
@@ -65,22 +65,19 @@ run_pki-key-template-show-kra_tests()
 	rlPhaseStartTest "pki_tey_template_show-002: verify when invalid key-template is provided key-template-show should fail"
 	local template=InvalidTemplate
 	rlRun "pki key-template-show $template > $temp_out1 2>&1" 255,1 "Executing pki key-template-show $template"
-	rlAssertGrep "Error: /usr/share/pki/key/templates/$template.xml (No such file or directory)" "$temp_out1"
+	rlAssertGrep "Error: /usr/share/pki/key/templates/$template.json (No such file or directory)" "$temp_out1"
```
Comment (edewata) [inline tests/dogtag/acceptance/cli-tests/pki-key-cli/pki-key-cli-template-show-kra.sh:68]: The tests under `tests/dogtag/acceptance` are probably not used anymore so if it's too difficult to update I'd say just leave it as is.

File: base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java [hunk 3]
```diff
@@ -52,18 +52,10 @@ public void createOptions() {
         option.setArgName("Passphrase");
         options.addOption(option);
 
-        option = new Option(null, "input-format", true, "Input format: xml (default), json");
-        option.setArgName("format");
-        options.addOption(option);
-
```
Comment (edewata) [inline base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java:58]: Will this cause existing scripts/automation that use `--input-format json` to fail? If so we probably should keep the option, but drop the `xml` and make `json` the default. If it's not causing a problem then this is fine.

File: base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java [hunk 4]
```diff
@@ -52,18 +52,10 @@ public void createOptions() {
...
         option = new Option(null, "input", true, "Location of the request template file.");
         option.setArgName("Input file path");
         options.addOption(option);
 
-        option = new Option(null, "output-format", true, "Output format: xml (default), json");
-        option.setArgName("format");
-        options.addOption(option);
-
```
Comment (edewata) [inline base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java:66]: Same thing as `--input-format`.

File: docs/changes/v11.1.0/Tools-Changes.adoc (hunk 1 above)
Comment (ckelleyRH) [reply docs/changes/v11.1.0/Tools-Changes.adoc:12]: Done.

File: tests/dogtag/acceptance/cli-tests/pki-key-cli/pki-key-cli-template-show-kra.sh (hunk 2 above)
Comment (ckelleyRH) [reply tests/dogtag/acceptance/cli-tests/pki-key-cli/pki-key-cli-template-show-kra.sh:68]: I have updated it anyway, but I have a PR somewhere to remove all the legacy tests which I'm tempted to just merge.

File: base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java (hunk 3 above)
Comment (ckelleyRH) [reply base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java]: Good point, I have reinstated the option defaulting to `JSON`. If `XML` is requested you will get errors, but requesting `JSON` is now backwards compatible.

File: base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java (hunk 4 above)
Comment (ckelleyRH) [reply base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java]: As above.

File: base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java [hunk 5]
```diff
@@ -101,10 +101,10 @@ public void execute(CommandLine cmd) throws Exception {
             String passphrase = cmd.getOptionValue("passphrase");
             String requestId = cmd.getOptionValue("requestID");
             String requestFile = cmd.getOptionValue("input");
-            String inputFormat = cmd.getOptionValue("input-format", "xml");
+            String inputFormat = cmd.getOptionValue("input-format", "json");
             String outputFilePath = cmd.getOptionValue("output");
             String outputDataFile = cmd.getOptionValue("output-data");
-            String outputFormat = cmd.getOptionValue("output-format", "xml");
+            String outputFormat = cmd.getOptionValue("output-format", "json");
```
Comment (edewata) [inline base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java:107]: Please also update the CLI help message, i.e. drop `xml` and make `json` the default.

File: base/server/python/pki/server/deployment/__init__.py [hunk 6]
```diff
@@ -1451,8 +1451,7 @@ def get_tps_connector(self, instance, subsystem):
             '-n', nickname,
             'tks-tpsconnector-show',
             '--host', self.mdict['pki_hostname'],
-            '--port', securePort,
-            '--output-format', 'json'
+            '--port', securePort
```
Comment (edewata) [inline base/server/python/pki/server/deployment/__init__.py:1454]: This code is calling `pki tks-tpsconnector-show` instead of `pki kra-key-recover`, so maybe we want to leave it for a different PR.

**Review by edewata — APPROVED** [review]
I have some more comments, but other than that it LGTM. Feel free to update and merge. Thanks!

File: base/server/python/pki/server/deployment/__init__.py (hunk 6 above)
Comment (ckelleyRH) [reply base/server/python/pki/server/deployment/__init__.py:1454]: Yeah good point, I'll reinstate.

File: base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java (hunk 5 above)
Comment (ckelleyRH) [reply base/tools/src/main/java/com/netscape/cmstools/kra/KRAKeyRetrieveCLI.java:107]: Ack.

**Comment by ckelleyRH** [comment]
Weird error building SSCEP but I don't think this caused the problem. Other CI failures are known issues.
